runtime/src/bank.rs

```diff
@@ -4935,6 +4951,21 @@ impl Bank {
             .round() as u64
     }
 
+    // Calculate cost of loaded accounts size in the same way heap cost is charged at
+    // rate of 8cu per 32K.
+    //
+    // Before feature `support_set_loaded_accounts_data_size_limit_ix` is enabled, or
+    // if user doesn't use compute budget ix `set_loaded_accounts_data_size_limit_ix`
+    // to set limit, `compute_budget.loaded_accounts_data_size_limit` is set to default
+    // limit of 64MB; which will convert to (64M/32K)*8CU = 16_000 CUs
+    //
+    fn calculate_loaded_accounts_data_size_cost(compute_budget: &ComputeBudget) -> u64 {
+        (compute_budget.loaded_accounts_data_size_limit as u64)
+            .saturating_div(32_u64.saturating_mul(1024))
+            .saturating_sub(1)
```

Comment:
why `-1`? div is already truncating the size (which it probably shouldn't?)

rounding up to the next 32k block makes more sense i think. so...
```
((64M+(32k-1))/32k)*8cu
```
rationale: you can't malloc one byte without allocating a whole page

Reply:
Ah, that makes more sense, thanks! I copy/paste code from runtime, I'll make a separate PR to fix same math in `programs/bpf_loader/src/lib.rs` as well